### ISSUE TITLE
@member of ES6 class showing as (inner) member of module / global

### ISSUE THREAD

**Issue by kruncher** [jsdoc3/jsdoc#1134]
Example:

**my-example/lib/Foo.js**

``` javascript
"use strict";

/** @module my-example/lib/Foo */

/**
 * An ES6 class.
 *
 * @alias module:my-example/lib/Foo
 */
class Foo {

    /**
     * My property.
     *
     * @member {string} _someProperty
     * @protected
     */

}
```

Given that the context of the property is non-ambiguous in the sense of it being a part of an ES6 class it seems overkill to have to use the `@memberof` tag. With this particular example `_someProperty` shows as an (inner) member of the module unless I specify the `@memberof` tag as shown here:

``` javascript
    /**
     * My property.
     *
     * @member {string} _someProperty
     * @memberof module:my-example/lib/Foo
     * @protected
     */
```

With the following example `_someOtherProperty` shows as a global:

**my-example/lib/OtherFoo.js**

``` javascript
"use strict";

/**
 * An ES6 class without specifying a module tag.
 */
class OtherFoo {

    /**
     * My other property.
     *
     * @member {string} _someOtherProperty
     * @protected
     */

}
```

**Comment by hegemonic** [comment]
Sorry, but that's how things work when you have a "virtual comment" that explicitly specifies the name of the symbol, as you're doing with `_someProperty` and `_someOtherProperty`. In those cases, JSDoc totally ignores the code and just looks at what's in the comment.

I was going to suggest the following as an alternative. However, this doesn't work correctly right now, which is a bug--the property gets the wrong longname:

```js
"use strict";

/** @module my-example/lib/Foo */

/**
 * An ES6 class.
 *
 * @alias module:my-example/lib/Foo
 */
class Foo {
    constructor() {
        /**
         * My property.
         *
         * @protected
         */
        this._someProperty = 'foo';
    }
}
```

I'm keeping this issue open to address that bug.

**Comment by Reinmar** [comment]
> With the following example _someOtherProperty shows as a global:

We've got the same problem. We need to explicitly define the full path to every property and method of every class... and those paths are super long now:

```
module:ckeditor5/engine/view/document#isFocused
```

So we hoped that we can use a shorter notation like:

```
@member {Boolean} #isFocused
```

And that JSDoc will automatically assign those properties to the class defined above. The same should happen with `{@link #isFocus}`. That's how it works in JSDuck which we used in the past and it was super convenient. 

We see that it works in JSDoc differently and we couldn't find any way to make those paths shorter, so we're now working on a plugin which will fix this for us. Let's see how it'll go.

But, TBH, I don't understand why it works in JSDoc this way. Perhaps it's a matter of focus. We're writing class based code, where a single module usually exports one class and every property is that class' property. But then I'd propose (too late of course ;)) to make a differentiation between how you document class members and module members, so based on a tag (e.g. `@member` vs `@export`) JSDoc would know whether to assign that thing to a class or module.

**Comment by hegemonic** [comment]
This may be a duplicate of #1182. More investigation required.

**Comment by hegemonic** [comment]
Not a duplicate of #1182, sadly.

**Comment by hegemonic** [comment]
Fixed on master and releases/3.5. The example I provided above is now processed correctly. The fix will be included in JSDoc 3.5.2.
